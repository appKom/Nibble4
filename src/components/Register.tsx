import React, { ChangeEvent, FC, useState } from "react";
import { Redirect } from "react-router-dom";
import styled from "styled-components";
import InputField from "atoms/InputField";
import { cancelRegisterRoute } from "utility/routes";
import Button, { ButtonColors } from "atoms/Button";

const Register: FC = () => {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [cancel, setCancel] = useState(false);

  const onUsernameChange = (event: ChangeEvent<HTMLInputElement>) =>
    setUsername(event.target.value);
  const onPasswordChange = (event: ChangeEvent<HTMLInputElement>) =>
    setPassword(event.target.value);

  if (cancel) return <Redirect to={cancelRegisterRoute} />;

  return (
    <Container>
      <h1>Registrer</h1>
      <InputField value={username} onChange={onUsernameChange} />
      <InputField
        value={password}
        onChange={onPasswordChange}
        type="password"
      />
      <Button>Registrer</Button>
      <Button buttonColor={ButtonColors.RED} onClick={() => setCancel(true)}>
        Tilbake
      </Button>
    </Container>
  );
};

const Container = styled.div`
  margin: auto;
  h1 {
    text-align: center;
  }
`;

export default Register;
