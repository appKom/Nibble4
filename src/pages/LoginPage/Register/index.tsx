import React, { ChangeEvent, FC, useContext, useState } from "react";
import { Redirect } from "react-router-dom";
import styled from "styled-components";
import InputField from "components/InputField";
import { cancelRegisterRoute } from "utility/routes";
import Button, { ButtonColors } from "components/Button";
import { handleRfid, registerUser } from "api/authorization";
import { StateContext } from "state/state";
import { setUser } from "state/actions";
import InfoBox, { InfoBoxTypes } from "./InfoBox";

type Props = {
  rfid: string;
};

const Register: FC<Props> = ({ rfid }: Props) => {
  const { dispatch } = useContext(StateContext);
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [badCredentials, setBadCredentials] = useState(false);
  const [cancel, setCancel] = useState(false);

  const onUsernameChange = (event: ChangeEvent<HTMLInputElement>) =>
    setUsername(event.target.value);

  const onPasswordChange = (event: ChangeEvent<HTMLInputElement>) =>
    setPassword(event.target.value);

  const onRegister = async () => {
    const response = await registerUser(username, password, rfid);
    if (response.ok) {
      const user = await handleRfid(rfid);
      if (user) dispatch(setUser(user));
    } else {
      setBadCredentials(true);
    }
  };

  const onCancel = () => setCancel(true);

  if (cancel) return <Redirect to={cancelRegisterRoute} />;

  return (
    <Container>
      <h1>Registrer</h1>
      <InfoBox>
        <p>
          Registrer deg med ditt <span>online.ntnu.no</span> brukernavn og
          passord.
        </p>
      </InfoBox>
      {badCredentials ? (
        <InfoBox type={InfoBoxTypes.WARNING}>
          Du skrev inn feil brukernavn eller passord. Prøv igjen på nytt.
        </InfoBox>
      ) : null}
      <InputField
        inputLabel="Brukernavn"
        value={username}
        onChange={onUsernameChange}
      />
      <InputField
        inputLabel="Passord"
        value={password}
        onChange={onPasswordChange}
        type="password"
      />
      <div>
        <Button buttonColor={ButtonColors.RED} onClick={onCancel}>
          Tilbake
        </Button>
        <Button onClick={onRegister}>Registrer</Button>
      </div>
    </Container>
  );
};

const Container = styled.div`
  display: flex;
  flex-direction: column;
  margin: auto;
  h1 {
    text-align: center;
  }
  div:last-child {
    display: flex;
    justify-content: space-between;
  }
  > * {
    margin: 0.5rem 0;
  }
`;

export default Register;
