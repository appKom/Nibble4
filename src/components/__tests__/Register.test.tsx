import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Register from "../Register";

afterEach(cleanup);
it("renders", () => {
  const { asFragment } = render(<Register rfid={123456789} />);
  expect(asFragment()).toMatchSnapshot();
});
