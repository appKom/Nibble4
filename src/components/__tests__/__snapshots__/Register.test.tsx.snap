// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="sc-hKgILt eLpZrX"
  >
    <h1>
      Registrer
    </h1>
    <div
      class="sc-dlfnbm etrdhU"
    >
      <p>
        <span>
          Registrer deg med ditt 
          <span>
            online.ntnu.no
          </span>
           brukernavn og passord.
        </span>
      </p>
    </div>
    <div
      class="sc-bdfBwQ gaVwxC"
    >
      <label>
        Brukernavn
      </label>
      <input
        id="Brukernavn"
        value=""
      />
    </div>
    <div
      class="sc-bdfBwQ gaVwxC"
    >
      <label>
        Passord
      </label>
      <input
        id="Passord"
        type="password"
        value=""
      />
    </div>
    <div>
      <button
        class="sc-gsTCUz buoxQL"
      >
        Tilbake
      </button>
      <button
        class="sc-gsTCUz dhzAWv"
      >
        Registrer
      </button>
    </div>
  </div>
</DocumentFragment>
`;
